### PR TITLE
KAFKA-18047: Add org.apache.kafka.automatic.config.providers to System Properties section

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: KAFKA-18047

Add documentation of `org.apache.kafka.automatic.config.providers` to System Properties section.
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by frankvicky** [comment]
Preview:

Screenshot from 2024-11-23 15-39-37

**Comment by frankvicky** [comment]
Hello @mimaison 
Could you please take a look at this PR when you have time? 
Many thanks! 🙇🏼

File: docs/configuration.html [hunk 1]
```diff
@@ -302,6 +302,22 @@ <h4><a id="org.apache.kafka.disallowed.login.modules"></a><a id="systempropertie
...
       </tbody></table>
     </li>
+    <li>
+      <h4><a id="org.apache.kafka.automatic.config.providers"></a><a id="systemproperties_org.apache.kafka.automatic.config.providers" href="#systemproperties_org.apache.kafka.automatic.config.providers">org.apache.kafka.automatic.config.providers</a></h4>
+      <p>This system property controls the automatic loading of ConfigProvider implementations in Apache Kafka. ConfigProviders are used to dynamically supply configuration values from sources such as files, directories, or environment variables. This property accepts a comma-separated list of ConfigProvider names. By default, all built-in ConfigProviders are enabled, including <b>FileConfigProvider</b>, <b>DirectoryConfigProvider</b>, and <b>EnvVarConfigProvider</b>.</p>
+      <p>If users want to disable all automatic ConfigProviders, they need to explicitly set the system property as shown below. Disabling automatic ConfigProviders is recommended in environments where configuration data comes from untrusted sources or where increased security is required. For more details, see <a href="https://nvd.nist.gov/vuln/detail/CVE-2024-31141">CVE-2024-31141</a>.</p>
```
Comment (mimaison) [inline docs/configuration.html]: Should we link to https://kafka.apache.org/cve-list#CVE-2024-31141 instead?

Comment (frankvicky) [reply docs/configuration.html]: It is ok for me.  
Currently, I use the link from `https://nvd.nist.gov/`, to align with the  
[org.apache.kafka.disallowed.login.modules ](https://kafka.apache.org/documentation/#systemproperties_org.apache.kafka.disallowed.login.modules)

If we decide to use the CVE link from the kafka-site, I suggest that we also update the link for `CVE-2023-25194.`

WDYT ?

Comment (mimaison) [reply docs/configuration.html]: I think it makes sense to point to our page which has a bit more details than the nist website. And our page has a link to nist.

Comment (frankvicky) [reply docs/configuration.html]: Make sense.
I have updated the link, PTAL 😺

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Comment by mimaison** [comment]
Not waiting for the CI to complete as this only changes `configuration.html`, merging to trunk

**Comment by mimaison** [comment]
Backported to 3.9 and 3.8

File: docs/configuration.html [hunk 2]
```diff
@@ -302,6 +302,22 @@ <h4><a id="org.apache.kafka.disallowed.login.modules"></a><a id="systempropertie
...
+      <p>If users want to disable all automatic ConfigProviders, they need to explicitly set the system property as shown below. Disabling automatic ConfigProviders is recommended in environments where configuration data comes from untrusted sources or where increased security is required. For more details, see <a href="https://kafka.apache.org/cve-list#CVE-2024-31141">CVE-2024-31141</a>.</p>
+      <pre><code class="language-bash">-Dorg.apache.kafka.automatic.config.providers=none</code></pre>
+      <p>To allow specific ConfigProviders, update the system property with a comma-separated list of ConfigProvider names. For example, to enable only the <b>EnvVarConfigProvider</b>, set the property as follows:</p>
+      <pre><code class="language-bash">-Dorg.apache.kafka.automatic.config.providers=env</code></pre>
```
Comment (chia7712) [inline docs/configuration.html:311]: It seems to me that =evn is incorrect. According to the source code, the correct value should be the class name. for example: `org.apache.kafka.common.config.provider.EnvVarConfigProvider` - I have opened https://issues.apache.org/jira/browse/KAFKA-18850 to fix it

**Pull request merged**